src/envs/screws.py

```diff
@@ -0,0 +1,509 @@
+    screw.
+    """
+    gripper_width: ClassVar[float] = 0.5
+    _receptacle_width: ClassVar[float] = 0.6
+    _receptacle_height: ClassVar[float] = 0.2
+    _screw_width: ClassVar[float] = 0.05
+    _screw_height: ClassVar[float] = 0.05
+    _magnetic_field_dist: ClassVar[float] = 0.5
+    _num_screw_clusters: ClassVar[int] = 3
+    num_screws_train: ClassVar[List[int]] = CFG.screws_num_screws_train
+    num_screws_test: ClassVar[List[int]] = CFG.screws_num_screws_test
+    # Reachable zone boundaries.
+    rz_x_lb = -5.0
+    rz_x_ub = 5.0
+    rz_y_lb = 0.0
+    rz_y_ub = 5.0
+
+    def __init__(self) -> None:
+        super().__init__()
+        # Types
+        self._screw_type = Type(
+            "screw", ["pose_x", "pose_y", "width", "height", "held"])
+        self._gripper_type = Type("gripper", ["pose_x", "pose_y", "width"])
+        self._receptacle_type = Type("receptacle",
+                                     ["pose_x", "pose_y", "width"])
+        # Predicates
+        self._ScrewPickupable = Predicate(
+            "ScrewPickupable", [self._gripper_type, self._screw_type],
+            self._ScrewPickupable_holds)
+        self._AboveReceptacle = Predicate(
+            "AboveReceptacle", [self._gripper_type, self._receptacle_type],
+            self._AboveReceptacle_holds)
+        self._HoldingScrew = Predicate("HoldingScrew", [self._screw_type],
+                                       self._HoldingScrew_holds)
+        self._ScrewInReceptacle = Predicate(
+            "ScrewInReceptacle", [self._screw_type, self._receptacle_type],
+            self._ScrewInReceptacle_holds)
+
+        # Options
+        self._MoveToScrew: ParameterizedOption = \
+            utils.SingletonParameterizedOption(
+            # variables: [robot, screw to pick up]
+            # params: []
+            "MoveToScrew",
+            self.MoveToScrew_policy,
+            types=[self._gripper_type, self._screw_type])
+        self._MoveToReceptacle: ParameterizedOption = \
+            utils.SingletonParameterizedOption(
+            # variables: [robot, receptacle]
+            # params: []
+            "MoveToReceptacle",
+            self.MoveToReceptacle_policy,
+            types=[self._gripper_type, self._receptacle_type])
+        self._MagnetizeGripper: ParameterizedOption = \
+            utils.SingletonParameterizedOption(
+            # variables: [robot]
+            # params: []
+            "MagnetizeGripper",
+            self.MagnetizeGripper_policy,
+            types=[self._gripper_type])
+        self._DemagnetizeGripper: ParameterizedOption = \
+            utils.SingletonParameterizedOption(
+            # variables: [robot]
+            # params: []
+            "DemagnetizeGripper",
+            self.DemagnetizeGripper_policy,
+            types=[self._gripper_type])
+
+        # Static objects (always exist no matter the settings).
+        self.robot = Object("robby", self._gripper_type)
```

Comment:
make private?

Reply:
also need in a test...

Reply:
you can get the robot from a state, like here: https://github.com/Learning-and-Intelligent-Systems/predicators/blob/master/tests/envs/test_doors_env.py#L55